### PR TITLE
Pretty printing paths

### PR DESCRIPTION
**Pull request by shrey-garg**

### CODE REVIEW

File: api/src/main/java/com/flipkart/poseidon/ds/trie/Trie.java [hunk 1]
```diff
@@ -157,30 +159,32 @@ private V get(TrieNode<K, V> node, int level, K[] keys) {
...
-    private void traverseAndPrint(TrieNode<K, V> node, String pathStr, String separator) {
+    private void traverseAndPrint(TrieNode<K, V> node, List<String> pathStr, String separator, List<List<String>> paths) {
         if (node == null) {
             return;
         }
 
-        StringBuilder strBuilder = new StringBuilder(pathStr);
-        if (!pathStr.endsWith(separator)) {
-            strBuilder.append(separator);
+        List<String> strBuilder = new ArrayList<>(pathStr);
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/ds/trie/Trie.java]: Changes this variable name?

File: api/src/main/java/com/flipkart/poseidon/ds/trie/Trie.java [hunk 2]
```diff
@@ -157,30 +159,32 @@ private V get(TrieNode<K, V> node, int level, K[] keys) {
...
+        if (pathStr.lastIndexOf(separator) != pathStr.size() - 1) {
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/ds/trie/Trie.java]: get() using index

File: api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java [hunk 3]
```diff
@@ -38,6 +39,7 @@
  */
 public abstract class APILegoSet extends PoseidonLegoSet {
     private static final Logger logger = getLogger(APILegoSet.class);
+    private final int methodLength = 7;
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java]: Captialize final variable name, use OPTIONS.length()

File: api/src/main/java/com/flipkart/poseidon/ds/trie/Trie.java (hunk 2 above)
Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/ds/trie/Trie.java]: get will result in an ArrayIndexOutOfBounds for the get(-1)

**Comment by shrey-garg** [comment]
changes done

**Pull request merged**